### PR TITLE
Badge API addition

### PR DESCRIPTION
**Pull request by BannerGames**
Hi. I'm here to pull request new methods related to the [Badge API](https://badges.roblox.com/docs#!/v1).

### **Additions**:
- New folder `badges` with two methods: **getBadgeInfo** and **updateBadgeInfo**
- New method in the `game` folder: **getGamesBadges**
- New method in the `user` folder: **getPlayerBadges**

Also, I've done the typings.

### CODE REVIEW

File: lib/user/getPlayerBadges.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      throw new Error('User is invalid or does not exist.')
+    }
+  })
+}
+
+exports.func = async (args) => {
+  if (isNaN(args.userId)) {
+    throw new Error('The provided User ID is not a number.')
+  };
+  if (![10, 25, 50, 100].includes(args.limit)) {
```
Comment (Neztore) [inline lib/user/getPlayerBadges.js:30]: Errors when limit is not provided

File: lib/game/getGameBadges.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  })
+}
+exports.func = async (args) => {
+  if (isNaN(args.universeId)) {
+    throw new Error('The provided Universe ID is not a number.')
+  };
+  if (![10, 25, 50, 100].includes(args.limit)) {
+    throw new Error('The allowed values are: 10, 25, 50 and 100.')
+  };
+  if (isNaN(args.limit)) {
```
Comment (Neztore) [inline lib/game/getGameBadges.js:32]: Seems redundant: You already check its one of the allowed numbers?

File: lib/game/getGameBadges.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    throw new Error('Limit is not a number.')
+  };
+  if (args.sortOrder.toLowerCase() !== 'asc' && args.sortOrder.toLowerCase() !== 'desc') {
```
Comment (Neztore) [inline lib/game/getGameBadges.js:35]: "Optional" but will error if not provided

**Comment by BannerGames** [comment]
I updated it

**Comment by BannerGames** [comment]
Also, I've updated the `setRank` method to feature the recently pull-requested change.

File: lib/badges/updateBadgeInfo.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  return getGeneralToken({ jar: jar }).then(xcrsf => {
+    return updateInfo(args.badgeId, name, description, enabled, xcrsf, jar)
+  })
+}
+exports.func = async (args) => {
+  if (isNaN(args.badgeId)) {
+    throw new Error('The provided Badge ID is not a number.')
+  }
+  if (typeof args.name !== 'string' || typeof args.description !== 'string') {
```
Comment (Neztore) [inline lib/badges/updateBadgeInfo.js:57]: Will error if name or description are not provided. They are marked as optional?

File: lib/badges/updateBadgeInfo.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  return getGeneralToken({ jar: jar }).then(xcrsf => {
+    return updateInfo(args.badgeId, name, description, enabled, xcrsf, jar)
+  })
+}
+exports.func = async (args) => {
+  if (isNaN(args.badgeId)) {
+    throw new Error('The provided Badge ID is not a number.')
+  }
+  if (args.name || args.description) {
```
Comment (Neztore) [inline lib/badges/updateBadgeInfo.js:57]: This means that if name or description are provided, **both** will be required. Is this intended behaviour?

Comment (BannerGames) [reply lib/badges/updateBadgeInfo.js:57]: So, you say I have to change it to `&&`?

Comment (Neztore) [reply lib/badges/updateBadgeInfo.js:57]: I would suggest separate conditionals entirely.
```js
 if (args.name) {
   // Check args.name
}
if (args.description) {
  //
}
```

**Pull request merged**